src/groupByKey.ts

```diff
@@ -1,7 +1,19 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key: string | number]: T[];
```

Comment:
Why do you need `number` type if you explicitly convert `itemKeyValue` to `string`